tests/refs/reflog/reflog.c

```diff
@@ -154,6 +154,59 @@ void test_refs_reflog_reflog__reading_the_reflog_from_a_reference_with_no_log_re
 	git_buf_free(&subtrees_log_path);
 }
 
+void test_refs_reflog_reflog__reading_a_reflog_with_invalid_format_returns_error(void)
+{
+	git_reflog *reflog;
+	const git_error *error;
+	const char *refname = "refs/heads/newline";
+	const char *refmessage =
+		"Reflog*message with a newline and enough content after it to pass the GIT_REFLOG_SIZE_MIN check inside reflog_parse.";
+	git_reference *ref;
+	git_oid id;
+	git_buf newline_log_path = GIT_BUF_INIT;
+	FILE *f;
+	int buf_length;
+	char *buf;
+	int i;
+
+	git_oid_fromstr(&id, current_master_tip);
+
+	/* create a new branch */
+	cl_git_pass(git_reference_create(&ref, g_repo, refname, &id, 1, refmessage));
+
+	/* corrupt the branch reflog by introducing a newline inside the reflog message (we replace '*' with '\n') */
+	git_buf_join_n(&newline_log_path, '/', 3, git_repository_path(g_repo), GIT_REFLOG_DIR, refname);
+	cl_assert((f = fopen(git_buf_cstr(&newline_log_path), "r+")) != NULL);
+	cl_assert(fseek(f, 0, SEEK_END) != -1);
+	buf_length = ftell(f);
+	cl_assert(fseek(f, 0, SEEK_SET) != -1);
+	cl_assert((buf = malloc(buf_length)) != NULL);
+	cl_assert(fread(buf, 1, buf_length, f) == buf_length);
+	for(i=0; i<buf_length; i++) {
+		if(buf[i] == '*') {
+			buf[i] = '\n';
+		}
+	}
```

Comment:
I'd recommend `strchr()` here as a more concise way to express this search. We should also assert that we did perform the replacement. otherwise something's gone quite wrong in the preparation of the test.